src/btree/bt_discard.c

```diff
@@ -531,3 +531,33 @@ __wt_free_update_list(WT_SESSION_IMPL *session, WT_UPDATE **updp)
     }
     *updp = NULL;
 }
+
+/*
+ * __wt_free_obsolete_updates --
+ *     Following a globally visible update, free any obsolete updates in the update chain. After a
+ *     globally visible update, no reader finds any updates. It is the responsibility of the caller
+ *     to lock the page before freeing the updates.
+ */
+void
+__wt_free_obsolete_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_UPDATE *visible_all_upd)
+{
+    WT_UPDATE *next, *upd;
+    size_t size;
+
+    size = 0;
+
+    next = visible_all_upd->next;
+    visible_all_upd->next = NULL;
```

Comment:
The old code had a comment here explaining why this update to a shared variable didn't need protection. It was ambiguous, since it didn't mention that it was only protecting against other threads that are also trimming obsolete updates, since readers rely on a different mechanism to avoid traversing past this point in the chaIn.

Could you add a new (better) comment in here please?

Reply:
I added a better comment to explain how the updates are protected.